swift/extractor/visitors/PatternVisitor.h

```diff
@@ -0,0 +1,34 @@
+#pragma once
+
+#include "swift/extractor/SwiftDispatcher.h"
+#include <swift/AST/ASTVisitor.h>
+
+namespace codeql {
+
+// swift code lacks default implementations of visitor for some entities. We can add those here
+// while we do not have yet all implemented. This is copy and pasted from the corresponding Expr
+// code in swift/AST/ASTVisitor.h
+template <typename ImplClass, typename PatternRetTy = void, typename... Args>
```

Comment:
The `typename PatternRetTy = void, typename... Args` look unnecessary? (Here and in `PatchedStmtVisitor`.)